### ISSUE TITLE
Global command and response fifos

### ISSUE THREAD

**Issue by raiguard** [mawww/kakoune#4363]
### Feature 

I would like to request the addition of two values: `kak_global_command_fifo` and `kak_global_response_fifo`. These would remain the same through the entire life of a kakoune session, and would allow for easy two-way communication between kakoune and a companion process.

### Usecase 

I would use these options in my [kak-loli](https://github.com/raiguard/kak-loli) project to remove the need to pass `kak_command_fifo` and `kak_response_fifo` in every call to the tool:

```sh
# Current method
$kak_opt_loli_cmd -i $kak_command_fifo -o $kak_response_fifo grep "$output"

# Potential method
$kak_opt_loli_cmd grep "$output"
```

**Comment by Screwtapello** [comment]
Kakoune could easily have a single `global_command_fifo` and `global_response_fifo` pair, but the moment two plugins tried to use it, they'd trip each other up in confusing and difficult-to-debug ways, so that seems like a bad idea.

A better plan would be to have a fifo pair *per plugin*, but then Kakoune needs to launch the plugin, potentially re-launch it if it crashes, and stop it when the Kakoune session exits. That's a more difficult design problem, explored a little in https://github.com/mawww/kakoune/issues/4127 although it still needs a bit of thought.

**Comment by raiguard** [comment]
Yeah, we mentioned proper co-process support on the Discord a while ago. I was thinking that a single global set of FIFOs would be fine because kak is entire synchronous, but that immediately falls apart when you call an external tool.

Perhaps adding some sort of command to "request" a set of FIFOs would work? Pass in a unique ID and it'll create FIFOs with that name (e.g. `loli` -> `kak_loli_command_fifo`), then you have known variable names to work with. If anyone else uses your variables, that's on them.

**Comment by lenormf** [comment]
>Perhaps adding some sort of command to "request" a set of FIFOs would work?

We already have `kak -p`, with its own communication protocol :)
